test/functional/ex_cmds/cd_spec.lua

```diff
@@ -269,3 +269,25 @@ for _, cmd in ipairs {'getcwd', 'haslocaldir'} do
   end)
 end
 
+-- Test what happens when a directory does not exist
+describe('Printing the working directory when it does not exist', function ()
+  local temp_dir = 'Xtest-functional-ex_cmds-cd_spec.temp'
+  before_each(function()
+    clear()
+    lfs.mkdir(temp_dir)
+  end)
+
+  after_each(function()
+    lfs.rmdir(temp_dir)
+  end)
+
+  it('returns an empty string', function()
+    execute('cd ' .. temp_dir)
+    helpers.wait()
+    lfs.rmdir(temp_dir)
+    local working_dir = cwd()
+    eq('', working_dir)
```

Comment:
I usually write such things as something like `eq(0, exc_exec('let wd = getcwd()'))`: this makes it obvious that I check for absence of exceptions and not rely on implementation details of various helper functions.